### ISSUE TITLE
UTF-8 BOM causes encode errors with pygmentize command

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#558]
(Original issue 851 created by EricFromCanada on 2013-02-21T00:21:43.655888+00:00)

The pygmentize command should remove the UTF-8 BOM when sending to terminal or a file. For example, `pygmentize tests/examplefiles/BOM.js` prints the file's contents without issue, but `pygmentize tests/examplefiles/BOM.js | less` returns
```
#!python
UnicodeEncodeError: 'ascii' codec can't encode character u'\ufeff' in position 13: ordinal not in range(128)
```
Also, `cat tests/examplefiles/BOM.js | pygmentize -l js` returns
```
#!python
UnicodeDecodeError: 'ascii' codec can't decode byte 0xef in position 0: ordinal not in range(128)
```
Specifying UTF-8 as the encoding still causes the BOM to appear as a character:

`pygmentize -O encoding='utf-8' tests/examplefiles/BOM.js | less` returns
```
#!js
<U+FEFF>/* There is a BOM at the beginning of this file. */
```

Occurs with Python 2.7.3.

**Comment by Anteru** [comment]
(Original issue 851 last updated on 2014-01-09T19:29:16.832832+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by amarquesbra on 2013-12-27T12:25:26.759760+00:00)

To contribute:


```
#!python

# less foo.xml

*** Error while highlighting:
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe2' in position 44: ordinal not in range(128)
   (file "/usr/lib64/python2.7/codecs.py", line 351, in write)
```
By curiosity only, using OpenSUSE.

Renderize a XML file with n lines but show only these:


```
#!xml

<?xml version="1.0" encoding="UTF-8"?>
<ListingDataFeed xmlns:xs="http://www.w3.org/2001/XMLSchema">
  <Listings>
    <Listing>
      <ListingID>278132897162110</ListingID>
      <Title>
foo.xml (END)

```
And the last line of this excerpt was the bug:


```
#!xml

# more foo.xml
<?xml version="1.0" encoding="UTF-8"?>
<ListingDataFeed xmlns:xs="http://www.w3.org/2001/XMLSchema">
  <Listings>
    <Listing>
      <ListingID>278132897162110</ListingID>
      <Title>Comercial loja - Bairro Setor Marista em Goiânia</Title>

```

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-01-09T19:29:16.828456+00:00)

Closes #851: remove BOM also with explicit encoding

→ <<cset 64a7c878a9bb>>
